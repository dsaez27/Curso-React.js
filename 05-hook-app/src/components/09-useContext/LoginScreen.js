import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const { setUser } = useContext(UserContext);

    const handleLogin = () => {
        setUser({
            id: 123,
            name: 'Daniel',
            email: 'dssh27@gmail.com'
        })
    }

    return (
        <>
            <h1>LoginScreen</h1>
            <hr />
            <button className='btn btn-primary' onClick={handleLogin}>
                Login
            </button>
        </>
    );
};
