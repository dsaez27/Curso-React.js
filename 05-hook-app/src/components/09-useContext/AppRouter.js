import React from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { AboutScreen } from './AboutScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <Switch>
                    <Route path="/" exact component={HomeScreen} />
                    <Route path="/about" exact component={AboutScreen} />
                    <Route path="/login" exact component={LoginScreen} />
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    )
}
