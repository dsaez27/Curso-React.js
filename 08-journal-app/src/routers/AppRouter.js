import React, { useEffect, useState } from 'react';
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import { AuthRouter } from './AuthRouter';
import { LoginScreen } from './../components/auth/LoginScreen';
import { JournalScreen } from './../components/journal/JournalScreen';
import { firebase } from '../firebase/firebaseConfig';
import { useDispatch } from 'react-redux';
import { login } from '../redux/actions/auth';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import { startLoadingNotes } from '../redux/actions/notes';

export const AppRouter = () => {
    const dispatch = useDispatch();
    const [checking, setChecking] = useState(true);
    const [isLoggedIn, setIsLoggedIn] = useState(false);

    useEffect(() => {
        firebase.auth().onAuthStateChanged(async (user) => {
            if (user?.uid) {
                dispatch(login(user.uid, user.displayName));
                setIsLoggedIn(true);

                dispatch(startLoadingNotes(user.uid));
            } else {
                setIsLoggedIn(false);
            }
            setChecking(false);
        });
    }, [dispatch, setChecking, setIsLoggedIn]);

    if (checking) {
        return <h1>Wait...</h1>;
    }

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute
                        path='/auth'
                        component={AuthRouter}
                        isLoggedIn={isLoggedIn}
                    />
                    <PrivateRoute
                        exact
                        path='/'
                        component={JournalScreen}
                        isLoggedIn={isLoggedIn}
                    />
                    <Redirect to='/auth/login' component={LoginScreen} />
                </Switch>
            </div>
        </Router>
    );
};
