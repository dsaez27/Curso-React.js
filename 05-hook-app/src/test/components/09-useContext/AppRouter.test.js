import React from 'react';
import { AppRouter } from './../../../components/09-useContext/AppRouter';
import { mount } from 'enzyme';
import { UserContext } from './../../../components/09-useContext/UserContext';

describe('Pruebas en el <AppRouter />', () => {

    const user = {
        id: 1,
        name: 'Daniel'
    }

    const wrapper = mount(
        <UserContext.Provider
            value={{
                user,
            }}>
            <AppRouter />
        </UserContext.Provider>
    );

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    
    
})
