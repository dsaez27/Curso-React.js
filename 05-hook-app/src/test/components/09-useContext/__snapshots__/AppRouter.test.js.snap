// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <AppRouter /> Debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-item nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
          <a
            aria-current={null}
            className="nav-item nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    Array [
      <h1>
        HomeScreen
      </h1>,
      <hr />,
      <pre>
        {
   "id": 1,
   "name": "Daniel"
}
      </pre>,
    ]
  </div>
</div>
`;
